### PR TITLE
dashboard test fix

### PR DESCRIPTION
**Pull request by pondrejk**
Fixing problem in UI dashboard tests

### CODE REVIEW

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_dashboard.py -k access_with_host_filter

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_dashboard.py -k access_with_host_filter

**Comment by pondrejk** [comment]
Ugh, passing locally

```
 pytest tests/foreman/ui/test_dashboard.py -k access_with_host_filter
================================================ test session starts =================================================                                       

tests/foreman/ui/test_dashboard.py .                                                                           [100%]

============================== 1 passed, 4 deselected, 6 warnings in 723.64s (0:12:03) ===============================
```

**Comment by ogajduse** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_dashboard.py -k access_with_host_filter

**Review by ogajduse — COMMENTED** [review]
I am not sure what exactly is happening there, but I see:
```
[D 230504 09:13:49 hosts:91] ip-10-0-0-1.acme.com executing command: yum install -y rabbit-1.5.6-1.noarch
[D 230504 09:13:50 hosts:93] ip-10-0-0-1.acme.com command result:
    stdout:
    Updating Subscription Management repositories.
    
    stderr:
    (107, b'Error: There are no enabled repositories in "/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d".\n')
    status: 1
```

```
[D 230504 09:13:31 hosts:91] ip-10-0-0-1.acme.com executing command: b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme  content-view info --id="4" '
[D 230504 09:13:33 hosts:93] ip-10-0-0-1.acme.com command result:
    stdout:
    Id:                           4
    Name:                         dEnrqwPgSB
    Label:                        dEnrqwPgSB
    Composite:                    false
    Description:                  
    Content Host Count:           0
    Solve Dependencies:           false
    Organization:                 xrWDkIUFUqoZ
    Yum Repositories:             
     1) Id:    1
        Name:  hxAYVLZjTr
        Label: hxAYVLZjTr
    Container Image Repositories: 
    
    OSTree Repositories:          
    
    Lifecycle Environments:       
     1) Id:   3
        Name: Library
     2) Id:   4
        Name: TTuCZJHAbR
    Versions:                     
     1) Id:        4
        Version:   1.0
        Published: 2023/05/04 09:13:20
    Components:                   
    
    Activation Keys:
    
    
    stderr:
    (0, b'')
    status: 0
```

```
[D 230504 09:13:01 hosts:91] ip-10-0-216-207.rhos-01.prod.psi.rdu2.redhat.com executing command: b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme  repository info --id="1" '
[D 230504 09:13:02 hosts:93] ip-10-0-216-207.rhos-01.prod.psi.rdu2.redhat.com command result:
    stdout:
    Id:                 1
    Name:               hxAYVLZjTr
    Label:              hxAYVLZjTr
    Description:        
    Organization:       xrWDkIUFUqoZ
    Red Hat Repository: no
    Content Type:       yum
    Mirroring Policy:   Content Only
    Url:                http://infra-podman.acme.com/fake_yum3
    Publish Via HTTP:   yes
    Published At:       https://ip-10-0-0-1.acme.com/pulp/content/xrWDkIUFUqoZ/Library/custom/toPiOSQYzP/hxAYVLZjTr/
    Relative Path:      xrWDkIUFUqoZ/Library/custom/toPiOSQYzP/hxAYVLZjTr
    Download Policy:    on_demand
    HTTP Proxy:         
        HTTP Proxy Policy: global_default_http_proxy
    Product:            
        Id:   1
        Name: DjILIRPYBD
    GPG Key:            
    
    Sync:               
        Status: Not Synced
    Created:            2023/05/04 09:12:59
    Updated:            2023/05/04 09:13:01
    Content Counts:     
        Packages:       0
        Source RPMS:    0
        Package Groups: 0
        Errata:         0
        Module Streams: 0
```

**Comment by ogajduse** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_dashboard.py -k access_with_host_filter

File: robottelo/host_helpers/repository_mixins.py [hunk 1]
```diff
@@ -691,13 +691,12 @@ def setup_content(
...
         if custom_product_name:
             subscription_names.append(custom_product_name)
-        activation_key = self.setup_activation_key(
-            org_id,
-            content_view['id'],
-            lce_id,
-            subscription_names=subscription_names,
-            override=override,
-        )
+        if self.satellite.is_sca_mode_enabled(org_id) is False:
```
Comment (ogajduse) [inline robottelo/host_helpers/repository_mixins.py:694]: minor suggestion
```suggestion
        if not self.satellite.is_sca_mode_enabled(org_id):
```

**Comment by pondrejk** [comment]
@ogajduse sorry for longer silence, the problem was apparently in custom repo not being enabled by default anymore, will retrigger prt

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_dashboard.py -k access_with_host_filter

**Review by lhellebr — APPROVED** [review]
LGTM, but why did you change the repo and package?  Are they somehow different from the original ones?

**Review by synkd — APPROVED** [review]

**Comment by omkarkhatavkar** [comment]
most stream UI tests are failing because of the UI navigation on the menu bar i.e. ``` './/div[@id="vertical-nav" or contains(@class, "nav-pf-vertical")]/ul'``` This reminds me @jyejare importance of the Sanity and adding gating to accept the stream build.

**Comment by pondrejk** [comment]
Hi, I didn't change the repo, just explicitely enabled it. Too bad about the stream ui issue but its unrelated to the changes from this PR, I have this passing locally against 6.14

File: tests/foreman/ui/test_dashboard.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 from nailgun.entity_mixins import TaskFailedError
 
 from robottelo.config import settings
-from robottelo.constants import FAKE_1_CUSTOM_PACKAGE
+from robottelo.constants import FAKE_7_CUSTOM_PACKAGE
```
Comment (lhellebr) [inline tests/foreman/ui/test_dashboard.py:25]: This is the change I was asking about.

File: tests/foreman/ui/test_dashboard.py [hunk 3]
```diff
@@ -186,23 +186,24 @@ def test_positive_task_status(session):
...
 @pytest.mark.parametrize(
     'repos_collection',
     [
         {
-            'distro': 'rhel7',
-            'SatelliteToolsRepository': {},
-            'YumRepository': {'url': settings.repos.yum_6.url},
-        },
+            'distro': 'rhel8',
+            'YumRepository': {'url': settings.repos.yum_3.url},
```
Comment (lhellebr) [inline tests/foreman/ui/test_dashboard.py:200]: Or this.

Comment (pondrejk) [reply tests/foreman/ui/test_dashboard.py:200]: ah, yes... iirc that is due to the change in the rhel version, there was some problem with repos.yum_6 though I don't recall what it was exactly

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_dashboard.py -k access_with_host_filter
env:
    ROBOTTELO_server__deploy_arguments__deploy_sat_version: '6.14.0'
    ROBOTTELO_server__deploy_arguments__deploy_snap_version: '9.0'

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/ui/test_dashboard.py -k access_with_host_filter
env:
    ROBOTTELO_server__deploy_arguments__deploy_sat_version: '6.14.0'
    ROBOTTELO_server__deploy_arguments__deploy_snap_version: '9.0'
```

**Review by lpramuk — APPROVED** [review]

**Pull request merged**